### DISCUSSION TITLE
Add-PnPSiteScriptPackage for pnp PowerShell - Does not work?

### DISCUSSION THREAD

**Discussion by wniemiGHUb** [pnp/powershell#2982] [Q&A]
### Discussed in https://github.com/pnp/powershell/discussions/2968

<div type='discussions-op-text'>

<sup>Originally posted by **wniemiGHUb** April  3, 2023</sup>
Has anyone been able to get the Add-PnPSiteScriptPackage command to work?  Documentation is very scarce.  I attempted to combine a simple .json site script file with 1 excel file.  I package them together into a package.zip and attempt to call the command using pnpOnline PowerShell:  

`Add-PnPSiteScriptPackage -Title "MP Projects Package" -Description "Major Projects" -ContentPath "c:\provision\package.zip" `

and I receive the same error every time - what am I missing here?  thanks,  Wes.

Add-PnPSiteScriptPackage : Unexpected character encountered while parsing value: P. Path '', line 0, position 0.
At line:1 char:1
+ Add-PnPSiteScriptPackage -Title "MP Projects Package" -Description "M ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Add-PnPSiteScriptPackage], PSInvalidOperationException
    + FullyQualifiedErrorId : InvalidOperation,AddSiteScriptPackage</div>

**Comment by Mlavieri** [comment]
Unfortunately it doesn't seem to be available in all environments yet.  

I was getting the same error as you when working with PnP PowerShell and while not in the official SharePoint Online PowerShell schema there is an Add-SPOSiteScriptPackage command available, which while it still doesn't work gives a more detailed error that the functionality is not supported.

![image](https://user-images.githubusercontent.com/1187519/231272191-14193fe6-48ab-4499-b2ed-6254c65aaa88.png)

**Reply by wniemiGHUb** [reply]
Thank you for the response.  It is interesting that your error message is different than mine.  Also, the official site script schema (https://github.com/microsoft/json-schemas/blob/main/sp/site-design-script-actions.schema.json) does display the option for the addFile source parameter which is to get the file from a package you compile with the Add-PnPSiteScriptPackage powershell command:

`"addFile": {
        "type": "object",
        "properties": {
          "verb": {
            "type": "string",
            "pattern": "^addFile$"
          },
          "destination": {
            "type": "string",
            "description": "The location in the library to copy the file to, including the filename, relative to the root of the library.",
            "examples": ["Training Documents/Accounting/First Day.pdf"]
          },
          "source": {
            "type": "string",
            "description": "Optional parameter. **The location in the site script package to copy the file from**, relative to the root directory. Provide either this or sourceUrl."
          },
          "sourceUrl": {
            "type": "string",
            "description": "Optional parameter. The location in the tenant to copy the file from, as an absolute url. Provide either this or source."
          }
        },
        "required": ["verb", "destination"],
        "additionalProperties": false
      },`

**Comment by dadu4net** [comment]
@wniemiGHUb i am looking into this command. But i cannot find anything about the contents of the zip file. Does anyone have an url for an example zip?
